lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -686,20 +719,40 @@ monitorMetadata
     -> GenesisParameters
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp DBLayer{..} = do
+monitorMetadata tr gp db@(DBLayer{..}) = do
     settings <- atomically readSettings
     manager <- newManager defaultManagerSettings
+
     let fetcher fetchStrategies = fetchFromRemote trFetch fetchStrategies manager
-    forever $ do
-        (refs, successes) <- case poolMetadataSource settings of
-            FetchNone -> pure ([], [])
-            FetchDirect -> fetchThem $ fetcher [identityUrlBuilder]
-            FetchSMASH (unSmashServer -> uri) -> fetchThem $ fetcher [registryUrlBuilder uri]
-
-        when (null refs || null successes) $ do
-            traceWith tr $ MsgFetchTakeBreak blockFrequency
-            threadDelay blockFrequency
+        loop getPoolMetadata = forever $ do
+            (refs, successes) <- getPoolMetadata
+            when (null refs || null successes) $ do
+                traceWith tr $ MsgFetchTakeBreak blockFrequency
+                threadDelay blockFrequency
+
+    case poolMetadataSource settings of
+        FetchNone -> loop (pure ([], [])) -- TODO: exit loop?
+        FetchDirect -> loop (fetchThem $ fetcher [identityUrlBuilder])
+        FetchSMASH (unSmashServer -> uri) -> do
+            tid <-
+                forkFinally
+                    (gcDelistedPools tr db
+                        (fetchDelistedPools trFetch (toDelistedPoolsURI uri) manager)
+                    )
+                    onExit
```

Comment:
**Request:** simplify this expression, as it is much longer than the [line limit](https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters).

I can think of two possibilities here:

1. Give a name to the sub-expression.
```suggestion
            let getDelistedPools =
                    fetchDelistedPools trFetch (toDelistedPoolsURI uri) manager
            tid <- forkFinally
                (gcDelistedPools tr db getDelistedPools)
                onExit
```

2. Use `flip` to avoid a multi-line parenthetic expression.
```suggestion
            tid <- flip forkFinally onExit
                $ gcDelistedPools tr db
                $ fetchDelistedPools trFetch (toDelistedPoolsURI uri) manager
```